spec/unit/puppet/provider/conf_setting/ruby_spec.rb

```diff
@@ -212,14 +212,106 @@ def validate_file(expected_content,tmpfile = tmpfile)
       validate_file(" test_key_1 : { setting1 : \"helloworld\" }\n", emptyfile)
     end
 
-    it "should be able to handle variables of any type" do
+    it "should be able to handle variables of boolean type" do
       resource = Puppet::Type::Hocon_setting.new(common_params.merge(
-          :setting => 'test_key_1.master', :value => true))
+          :setting => 'test_key_1.master', :value => false))
       provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(false)
+    end
+
+    it "should be able to handle variables of integer type" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => 12))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(12)
+    end
+
+    it "should be able to handle variables of float type" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => 12.24))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(12.24)
+    end
+
+    it "should be able to handle arrays" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => [1, 2, 3, 4]))
+      provider = described_class.new(resource)
+      provider.create
       expect(provider.exists?).to be true
-      expect(provider.value).to eql(true)
+      expect(provider.value).to eql([1, 2, 3, 4])
     end
 
+    it "should be able to handle maps" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => {"a" => 1, "b" => 2}))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql({"a" => 1, "b" => 2})
```

Comment:
This test should cover that use case, as the `create` method handles adding the set value into the file and the `value` method works by using the HOCON API to grab the value of the desired setting from the config file.